go-controller/pkg/factory/handler.go

```diff
@@ -192,19 +194,28 @@ func (i *informer) removeHandler(handler *Handler) {
 
 		intInf.Lock()
 		defer intInf.Unlock()
-		removed := 0
+		removed := false
+		// track overall how many handlers this internal informer has
+		numHandlers := 0
```

Comment:
unrelated but what's the purpose of 
```

type WatchFactory struct {
	handlerCounter *handlerCounter
```
handler counter in the wf which we seem to also be using to count the number of handlers? I guess that is total across all informers?

Reply:
i guess it could probably removed, it doesnt look like its being used. It makes more sense to have the counter on the internal informer, since that is the thing we need to lock to protect write access